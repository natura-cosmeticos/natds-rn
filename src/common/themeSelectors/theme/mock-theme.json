{
  "avatarSizes": {
    "tiny": {
      "size": 24,
      "fontSize": ".625rem"
    },
    "small": {
      "size": 32,
      "fontSize": ".875rem"
    },
    "standard": {
      "size": 40,
      "fontSize": "1rem"
    },
    "large": {
      "size": 48,
      "fontSize": "1rem"
    },
    "huge": {
      "size": 80,
      "fontSize": "1.5rem"
    }
  },
  "buttonSizes": {
    "small": {
      "paddingTop": 8,
      "paddingRight": 8,
      "paddingBottom": 8,
      "paddingLeft": 8,
      "minHeight": 32
    },
    "medium": {
      "paddingTop": 12,
      "paddingRight": 16,
      "paddingBottom": 12,
      "paddingLeft": 16,
      "minHeight": 40
    },
    "large": {
      "paddingTop": 16,
      "paddingRight": 16,
      "paddingBottom": 16,
      "paddingLeft": 16,
      "minHeight": 48
    }
  },
  "fontSize": {
    "h1": {
      "fontSize": "6rem",
      "fontWeight": 300
    },
    "h2": {
      "fontSize": "3.75rem",
      "fontWeight": 300
    },
    "h3": {
      "fontSize": "3rem",
      "fontWeight": 400
    },
    "h4": {
      "fontSize": "2.125rem",
      "fontWeight": 400
    },
    "h5": {
      "fontSize": "1.5rem",
      "fontWeight": 400
    },
    "h6": {
      "fontSize": "1.25rem",
      "fontWeight": 600
    },
    "subtitle1": {
      "fontSize": "1rem",
      "fontWeight": 400
    },
    "subtitle2": {
      "fontSize": "0.875rem",
      "fontWeight": 500
    },
    "body1": {
      "fontSize": "1rem",
      "fontWeight": 400
    },
    "body2": {
      "fontSize": "0.875rem",
      "fontWeight": 400
    },
    "button": {
      "fontSize": "0.875rem",
      "fontWeight": 600
    },
    "caption": {
      "fontSize": "0.75rem",
      "fontWeight": 400
    },
    "overline": {
      "fontSize": "0.75rem",
      "fontWeight": 400
    }
  },
  "iconSizes": {
    "micro": 16,
    "tiny": 24,
    "small": 32,
    "standard": 48,
    "medium": 64,
    "large": 72,
    "largex": 96,
    "largexx": 128,
    "huge": 144,
    "hugex": 192,
    "hugexx": 256
  },
  "opacity": {
    "12": 0.12,
    "16": 0.16,
    "24": 0.24,
    "32": 0.32,
    "48": 0.48,
    "56": 0.56,
    "64": 0.64,
    "80": 0.8,
    "00": 0,
    "04": 0.04,
    "08": 0.08
  },
  "overlay": {
    "solidBlack": 0.56
  },
  "radius": {
    "none": 0,
    "circle": "50%",
    "small": 2,
    "medium": 4,
    "large": 8
  },
  "sizes": {
    "none": 0,
    "micro": 4,
    "tiny": 8,
    "small": 16,
    "standard": 24,
    "semi": 32,
    "semix": 40,
    "medium": 48,
    "mediumx": 56,
    "large": 64,
    "largex": 72,
    "largexx": 80,
    "largexxx": 88,
    "huge": 96,
    "hugex": 128,
    "hugexx": 144,
    "hugexxx": 192,
    "veryhuge": 256
  },
  "spacing": {
    "spacing": 8,
    "spacingNone": 0,
    "spacingMicro": 4,
    "spacingTiny": 8,
    "spacingSmall": 16,
    "spacingStandard": 24,
    "spacingSemi": 32,
    "spacingLarge": 48,
    "spacingHuge": 64
  },
  "colorTokens": {
    "colorPrimary": "#F4AB34",
    "colorOnPrimary": "#333333",
    "colorPrimaryLight": "#FEFDE8",
    "colorOnPrimaryLight": "#333333",
    "colorPrimaryDark": "#EF8426",
    "colorOnPrimaryDark": "#333333",
    "colorSecondary": "#FF6B0B",
    "colorOnSecondary": "#000000",
    "colorSecondaryLight": "#FFF8E1",
    "colorOnSecondaryLight": "#000000",
    "colorSecondaryDark": "#FF5808",
    "colorOnSecondaryDark": "#000000",
    "colorBackground": "#FAFAFA",
    "colorOnBackground": "#333333",
    "colorSurface": "#FFFFFF",
    "colorOnSurface": "#333333",
    "colorHighlight": "#000000",
    "colorHighEmphasis": "#333333",
    "colorMediumEmphasis": "#777777",
    "colorLowEmphasis": "#BBBBBB",
    "colorLink": "#227BBD",
    "colorOnLink": "#FFFFFF",
    "colorSuccess": "#569A32",
    "colorOnSuccess": "#FFFFFF",
    "colorWarning": "#FCC433",
    "colorOnWarning": "#333333",
    "colorAlert": "#E74627",
    "colorOnAlert": "#FFFFFF"
  },
  "elevation": {
    "1": {
      "elevation": 1,
      "shadowColor": "#000000",
      "shadowOffset": { "height": 1, "width": 0 },
      "shadowOpacity": 0.14,
      "shadowRadius": 1
    },
    "2": {
      "elevation": 2,
      "shadowColor": "#000000",
      "shadowOffset": { "height": 2, "width": 0 },
      "shadowOpacity": 0.14,
      "shadowRadius": 2
    },
    "3": {
      "elevation": 3,
      "shadowColor": "#000000",
      "shadowOffset": { "height": 3, "width": 0 },
      "shadowOpacity": 0.14,
      "shadowRadius": 4
    },
    "4": {
      "elevation": 4,
      "shadowColor": "#000000",
      "shadowOffset": { "height": 4, "width": 0 },
      "shadowOpacity": 0.14,
      "shadowRadius": 5
    },
    "5": {
      "elevation": 5,
      "shadowColor": "#000000",
      "shadowOffset": { "height": 6, "width": 0 },
      "shadowOpacity": 0.14,
      "shadowRadius": 10
    },
    "6": {
      "elevation": 6,
      "shadowColor": "#000000",
      "shadowOffset": { "height": 8, "width": 0 },
      "shadowOpacity": 0.14,
      "shadowRadius": 10
    },
    "7": {
      "elevation": 7,
      "shadowColor": "#000000",
      "shadowOffset": { "height": 9, "width": 0 },
      "shadowOpacity": 0.14,
      "shadowRadius": 12
    },
    "8": {
      "elevation": 8,
      "shadowColor": "#000000",
      "shadowOffset": { "height": 12, "width": 0 },
      "shadowOpacity": 0.14,
      "shadowRadius": 17
    },
    "9": {
      "elevation": 9,
      "shadowColor": "#000000",
      "shadowOffset": { "height": 16, "width": 0 },
      "shadowOpacity": 0.14,
      "shadowRadius": 24
    },
    "10": {
      "elevation": 10,
      "shadowColor": "#000000",
      "shadowOffset": { "height": 24, "width": 0 },
      "shadowOpacity": 0.14,
      "shadowRadius": 38
    }
  }
}
