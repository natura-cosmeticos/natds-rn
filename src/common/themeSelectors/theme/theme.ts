import { tokens as styleTokens } from '@naturacosmeticos/natds-styles';
import { IColorThemeTokens } from '@naturacosmeticos/natds-styles/dist/tokens/colors/themeBound/IColorThemeTokens';
import { IElevation } from '@naturacosmeticos/natds-styles/dist/tokens/elevation/IElevation';

type Tokens = typeof styleTokens;
type TokensWithoutColor = Omit<Tokens, 'colors'>
type TokensWithoutColorTokens = Omit<TokensWithoutColor, 'colorTokens'>
type TokensWithoutElevation = Omit<TokensWithoutColorTokens, 'elevation'>

interface ElevationProps {
  elevation: number
  shadowColor: string
  shadowOffset: { height: number, width: number }
  shadowOpacity: number
  shadowRadius: number
}

type Elevation = Record<keyof IElevation, ElevationProps>


export type Theme = TokensWithoutElevation
  & {
    colorTokens: IColorThemeTokens,
    elevation: Elevation | {}
  }

type ThemeMode = 'dark' | 'light';

const parseElevation = (elevation: string) => {
  const regEx = /(\d+(\.\d+)?)/g;
  const matches = elevation.match(regEx);

  if (matches) {
    const [width, height, blurRadius, , r, g, b, alpha] = matches;

    const toUnit = (prop: string) => parseInt(prop, 0);

    return {
      elevation: 2,
      shadowColor: `rgb(${r}, ${g}, ${b})`,
      shadowOffset: { height: toUnit(height), width: toUnit(width) },
      shadowOpacity: parseFloat(alpha),
      shadowRadius: toUnit(blurRadius),
    };
  }

  return {};
};

const buidElevation = (elevation: IElevation) => {
  const nativeElevation = {};

  Object
    .keys(elevation)
    .forEach((key) => { nativeElevation[key] = parseElevation(elevation[key]); });

  return nativeElevation;
};

export const buildTheme = (tokens: Tokens, brand: string, mode: ThemeMode) => {
  const colorThemeName = `${brand}${mode === 'light' ? 'Light' : 'Dark'}ColorTokens`;
  const localTokens = { ...tokens };

  delete localTokens.colors;

  const theme: Theme = {
    ...localTokens,
    colorTokens: localTokens.colorTokens[colorThemeName],
    elevation: buidElevation(localTokens.elevation),
  };

  return theme;
};

interface ThemeSelector {
  (theme: Theme): string | undefined
}

export const getTheme = (selector: ThemeSelector) => (
  { theme }: { theme: Theme },
) => selector(theme);
