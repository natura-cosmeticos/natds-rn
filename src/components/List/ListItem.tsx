import React, { useState, ReactElement } from 'react';
import { LayoutChangeEvent, View } from 'react-native';
import { Theme } from '../../common/themeSelectors';
import { TouchableRipple, TouchableRippleProps } from '../TouchableRipple/TouchableRipple';
import { ListItem as ListItemComponent } from './ListItem.styles';

export interface ListItemProps {
  /**
   * The children list item elements
   */
  children: ReactElement;
  /**
   * Controls the element selected styles
   */
  selected?: boolean;
  /**
   * onPress event handler
   */
  onPress?(): void;
  /**
   * Id for testing
   */
  testID?: string;
  /**
   * App's theme
   */
  theme?: Theme;
}

export const getRippleSize = (
  event: LayoutChangeEvent,
  setSize: React.Dispatch<React.SetStateAction<number>>,
) => {
  const { layout: { width, height } } = event.nativeEvent;
  const biggerSide = width >= height ? width : height;

  setSize((biggerSide / 2));
};

const ListWrapper = ({ onPress, children, ...props }: TouchableRippleProps) => {
  if (onPress) {
    return (
      <TouchableRipple onPress={onPress} {...props}>
        {children}
      </TouchableRipple>
    );
  }

  return (
    <View>
      {children}
    </View>
  );
};

export const ListItem = ({
  children,
  onPress,
  testID = 'list-item',
  ...rest
}: ListItemProps) => {
  const [size, setSize] = useState(0);

  return (
    <ListWrapper
      color="highlight"
      hideOverflow
      size={size}
      onPress={onPress}
      testID={`${testID}-wrapper`}
    >
      <ListItemComponent
        testID={testID}
        onLayout={event => onPress && getRippleSize(event, setSize)}
        onPress={onPress}
        {...rest}
      >
        {children}
      </ListItemComponent >
    </ListWrapper >
  );
};
