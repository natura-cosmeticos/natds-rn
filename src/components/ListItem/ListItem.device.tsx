import { storiesOf } from '@storybook/react-native'
import {
  Default,
  Dividers,
  RippleFeedback,
  SelectionFeedback
} from './ListItem.stories'

storiesOf('ListItem', module)
  .add('Default', Default)
  .add('FeedbackRipple', RippleFeedback)
  .add('FeedbackSelection', SelectionFeedback)
  .add('Dividers', Dividers)
