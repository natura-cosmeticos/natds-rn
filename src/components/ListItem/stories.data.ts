export const data = [
  {
    key: 'item1',
    title: 'A Game of Thrones ',
  },
  {
    key: 'item2',
    title: 'A Clash of Kings',
  },
  {
    key: 'item3',
    title: 'A Storm of Swords',
  },
  {
    key: 'item4',
    title: 'A Feast for Crows',
  },
  {
    extraInfo: {
      author: 'George R.R. Martin',
      rating: 10,
      releaseDate: '2000',
    },
    key: 'item5',
    title: 'A Dance with Dragons',
  },
  {
    key: 'item6',
    title: 'The Winds of Winter',
    unreleased: true,
  },
  {
    key: 'item7',
    title: 'A Dream of Spring',
    unreleased: true,
  },
];
