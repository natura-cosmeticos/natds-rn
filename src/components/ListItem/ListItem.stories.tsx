/* eslint-disable max-lines */
/* eslint-disable no-console */
import React, { useState } from 'react'
import { View, FlatList, Text } from 'react-native'
import { Card } from '../Card'
import { Divider } from '../Divider'
import { ListItem, ListItemFeedback } from './ListItem'
import { Checkbox } from '../Checkbox'
import { TextWithTheme } from '../../common/HelperComponents/ThemeHelper.styles'
import { data } from './stories.data'
import { StoryContainer } from '../../common/HelperComponents/StoryContainer'

export default {
  component: ListItem,
  parameters: {
    componentSubtitle: 'List items are rows of information that can have user interactions or not'
  },
  title: 'Components|List'
}

export const Default = () => (
  <StoryContainer title="Standard">
    <FlatList
      data={data}
      renderItem={({ item }) => (
        <ListItem>
          <View style={{ padding: 16 }}>
            <TextWithTheme>{item.title}</TextWithTheme>
          </View>
        </ListItem>
      )}
    />
  </StoryContainer>
)

export const RippleFeedback = () => (
  <StoryContainer title="Feedback Ripple">
    <FlatList
      data={data}
      renderItem={({ item }) => (
        <ListItem feedback="ripple" onPress={() => console.log('I was pressed')}>
          <View style={{ padding: 16 }}>
            <TextWithTheme>{item.title}</TextWithTheme>
          </View>
        </ListItem>
      )}
    />
  </StoryContainer>
)

export const SelectionFeedback = () => {
  const [selected, setSelected] = useState('')

  const renderItem = ({ item }) => {
    const listItemProps = {
      feedback: 'selection' as ListItemFeedback,
      selected: item.key === selected,
      ...(item.unreleased ? {} : { onPress: () => setSelected(item.key) })
    }

    return (
      <ListItem {...listItemProps}>
        <View style={{
          flexDirection: 'row',
          justifyContent: 'space-between',
          padding: 16
        }}
        >
          <TextWithTheme>{item.title}</TextWithTheme>
          <Checkbox
            selected={item.key === selected}
            onPress={() => setSelected(item.key)}
            disabled={item.unreleased}
          />
        </View>
      </ListItem>
    )
  }

  return (
    <StoryContainer title="Feedback Selection">
      <FlatList
        ItemSeparatorComponent={Divider}
        data={data}
        renderItem={renderItem}
        extraData={selected}
      />
    </StoryContainer>
  )
}

export const Dividers = () => {
  const renderExtraInfo = (extraInfo) => {
    if (!extraInfo) return null

    const extraInfoData = Object
      .keys(extraInfo)
      .map((item, index) => (
        <View key={extraInfo[item].key} style={{ backgroundColor: '#fafafa' }}>
          <ListItem>
            <View style={{
              flexDirection: 'row',
              justifyContent: 'space-between',
              paddingHorizontal: 32,
              paddingVertical: 16
            }}
            >
              <TextWithTheme>{item}</TextWithTheme>
              <TextWithTheme>{extraInfo[item]}</TextWithTheme>
            </View>
          </ListItem>
          {
            index < Object.keys(extraInfo).length - 1
              ? <Divider type="inset" />
              : null
          }
        </View>
      ))

    return extraInfoData
  }

  return (
    <StoryContainer title="Dividers">
      <View style={{
        flex: 1,
        paddingHorizontal: 10,
        paddingVertical: 20
      }}
      >
        <Card>
          <FlatList
            ItemSeparatorComponent={Divider}
            data={data}
            renderItem={({ item }) => (
              <>
                <ListItem onPress={() => console.log('I was pressed')}>
                  <View style={{ padding: 16 }}>
                    <Text>{item.title}</Text>
                  </View>
                </ListItem>
                {renderExtraInfo(item.extraInfo)}
              </>
            )}
          />
        </Card>
      </View>
    </StoryContainer>
  )
}
