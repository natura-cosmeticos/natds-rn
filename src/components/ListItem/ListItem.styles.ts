import styled from 'styled-components/native'
import {
  getColorPrimary,
  getOpacityMediumLow,
  buildColorWithOpacity,
  getSize,
  getTheme
} from '../../common/themeSelectors'
import { ListItemProps } from './ListItem'

const backgroundStyles = ({ selected, theme }: Required<ListItemProps>) => (
  selected
    ? buildColorWithOpacity(getColorPrimary, getOpacityMediumLow, theme)
    : 'transparent'
)

const getSizeMedium = getTheme((theme) => getSize(theme, 'medium'))

export const ListItem = styled.View<ListItemProps>`
  background: ${backgroundStyles};
  min-height: ${getSizeMedium}px;
`
