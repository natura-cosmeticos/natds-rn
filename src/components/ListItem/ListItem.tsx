import React, { ReactElement } from 'react'
import { View, TouchableWithoutFeedback } from 'react-native'
import { Theme } from '../../common/themeSelectors'
import { TouchableRipple, TouchableRippleProps } from '../TouchableRipple/TouchableRipple'
import { ListItem as ListItemComponent } from './ListItem.styles'

export type ListItemFeedback = 'ripple' | 'selection'

export interface ListItemProps {
  /**
   * The children list item elements
   */
  children: ReactElement;
  /**
   * Controls the element feedback type `ripple` | `selection`
   * @Default ripple
   */
  feedback?: ListItemFeedback;
  /**
   * Controls the element selected styles when feedback type is `selection`
   * @Default false
   */
  selected?: boolean;
  /**
   * onPress event handler
   */
  onPress?(): void;
  /**
   * Id for testing
   */
  testID?: string;
  /**
   * App's theme
   */
  theme?: Theme;
}

type ListWrapper = TouchableRippleProps & Pick<ListItemProps, 'feedback'>

const ListWrapper = ({
  onPress, children, feedback, ...props
}: ListWrapper) => {
  if (!onPress) {
    return (
      <View>
        {children}
      </View>
    )
  }

  if (feedback === 'ripple') {
    return (
      <TouchableRipple onPress={onPress} {...props}>
        {children}
      </TouchableRipple>
    )
  }

  return (
    <TouchableWithoutFeedback onPress={onPress} {...props}>
      {children}
    </TouchableWithoutFeedback>
  )
}

export const ListItem = ({
  children,
  onPress,
  testID = 'list-item',
  feedback = 'ripple',
  ...rest
}: ListItemProps) => (
  <ListWrapper
    color="highlight"
    hideOverflow
    onPress={onPress}
    testID={`${testID}-wrapper`}
    feedback={feedback}
  >
    <ListItemComponent
      testID={testID}
      onPress={onPress}
      {...rest}
    >
      {children}
    </ListItemComponent>
  </ListWrapper>
)
