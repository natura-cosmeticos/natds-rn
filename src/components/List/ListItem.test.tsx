import '@testing-library/jest-native/extend-expect';
import React from 'react';
import { render, fireEvent } from '@testing-library/react-native';
import { ThemeProvider } from 'styled-components/native';
import theme from '../../common/themeSelectors/theme/mock-theme.json';

import { ListItem, ListItemProps } from './ListItem';

jest.mock('../../common/themeSelectors', () => ({
  getColorBackground: () => '#FFFFFF',
  getColorByName: () => '#FFFFFF',
  getColorHighEmphasis: () => '#FAF3E3',
  getColorLowEmphasis: () => '#FEEEEF',
  getColorPrimary: () => '#FFFFFF',
  getSize: () => 50,
  getSpacingTiny: () => 10,
}));

jest.mock(
  'react-native/Libraries/Components/Touchable/TouchableOpacity.js',
  () => 'TouchableOpacity',
);

const renderList = (fn, props: ListItemProps) => (
  fn(
    <ThemeProvider theme={theme}>
      <ListItem {...props} />
    </ThemeProvider>,
  )
);

const props: ListItemProps = {
  active: false,
  divider: true,
  hideIconLeft: true,
  hideIconRight: true,
  onPress: jest.fn(),
  theme,
  title: 'Test',
};

describe('ListItem component', () => {
  it('Should render ListItem component correctly', () => {
    const listItem = renderList(render, props).asJSON();

    expect(listItem).toMatchSnapshot('ListItem snapshot');
  });

  it('Should throw exception when title prop is not provided', () => {
    expect(() => renderList(render, {
      ...props,
      title: '',
    })).toThrow(
      'Title should not be an empty string',
    );
  });

  it('Should render ListItem component with icon', () => {
    const { getByTestId } = renderList(render, {
      ...props,
      hideIconLeft: false,
      iconLeft: 'filled-action-add',
    });
    const iconListItem = getByTestId('listId-icon-left');

    expect(iconListItem).toBeTruthy();
  });

  it('Should call the given onPress function', () => {
    const onPressMock = jest.fn();

    const { getByTestId } = renderList(render, {
      ...props,
      onPress: onPressMock,
    });
    const listItem = getByTestId('listId-title');

    fireEvent.press(listItem);

    expect(onPressMock).toHaveBeenCalled();
  });

  it('Should not call the given onPress function when list is disabled', () => {
    const onPressMock = jest.fn();

    const { getByTestId } = renderList(render, {
      ...props,
      disabled: true,
      onPress: onPressMock,
    });
    const listItem = getByTestId('listId-title');

    fireEvent.press(listItem);

    expect(onPressMock).not.toHaveBeenCalled();
  });

  it('Should call the given icon onPress function', () => {
    const onPressLeftMock = jest.fn();

    const { getByTestId } = renderList(render, {
      ...props,
      hideIconLeft: false,
      iconLeft: 'filled-action-add',
      onPressLeft: onPressLeftMock,
    });
    const listItem = getByTestId('listId-icon-left');

    fireEvent.press(listItem);

    expect(onPressLeftMock).toHaveBeenCalled();
  });
});
