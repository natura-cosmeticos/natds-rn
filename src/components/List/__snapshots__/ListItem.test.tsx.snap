// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ListItem component Should render ListItem component correctly: ListItem snapshot 1`] = `
<View
  collapsable={true}
  pointerEvents="box-none"
  style={
    Object {
      "flex": 1,
    }
  }
>
  <View>
    <View
      accessibilityRole="header"
      accessible={true}
      active={false}
      style={
        Array [
          Object {
            "backgroundColor": "#FFFFFF29",
            "flexBasis": "auto",
            "flexGrow": 1,
            "flexShrink": 1,
            "paddingBottom": 0,
            "paddingLeft": 0,
            "paddingRight": 0,
            "paddingTop": 0,
          },
        ]
      }
      testID="listId"
    >
      <View
        hideIconLeft={true}
        hideIconRight={true}
        style={
          Array [
            Object {
              "alignItems": "center",
              "bottom": 0,
              "justifyContent": "center",
              "left": 0,
              "right": 0,
              "top": 0,
            },
          ]
        }
      >
        <TouchableOpacity
          style={
            Array [
              Object {
                "alignSelf": "stretch",
                "flexBasis": 0,
                "flexGrow": 1,
                "flexShrink": 1,
                "justifyContent": "center",
              },
              Object {
                "flexBasis": "auto",
              },
            ]
          }
          testID="listId-title"
        >
          <Text
            accessibilityLabel="Test"
            style={
              Array [
                Object {
                  "color": "#FAF3E3",
                  "flexBasis": "auto",
                  "flexGrow": 1,
                  "flexShrink": 1,
                  "fontSize": 16,
                  "justifyContent": "center",
                  "marginLeft": 0,
                  "paddingBottom": 12,
                  "paddingLeft": 16,
                  "paddingTop": 12,
                },
              ]
            }
          >
            Test
          </Text>
        </TouchableOpacity>
      </View>
    </View>
    <View>
      <View
        style={
          Array [
            Object {
              "backgroundColor": "#FEEEEF",
              "height": 1,
              "width": "100%",
            },
          ]
        }
      />
    </View>
  </View>
</View>
`;
